lib/logstash/outputs/elasticsearch/http_client/pool.rb

```diff
@@ -151,15 +151,48 @@ def sniff!
     ES2_SNIFF_RE_URL  = /([^\/]*)?\/?([^:]*):([0-9]+)/
     # Sniffs and returns the results. Does not update internal URLs!
     def check_sniff
-      url, resp = perform_request(:get, '_nodes')
+      url, resp = perform_request(:get, '_nodes/http')
```

Comment:
Also, it's not a behavioral change, before we had code to just ignore nodes without an HTTP endpoint. This is a bit more resource efficient since we don't need to receive nodes without HTTP. We can't do anything with those nodes anyway :)